src/runtime/components/nuxt-picture.ts

```diff
@@ -94,23 +92,25 @@ export default defineComponent({
       }
     })
 
-    return () => h('picture', { key: sources.value[0].src }, [
-      ...(sources.value?.[1]
-        ? [h('source', {
-            type: sources.value[1].type,
-            sizes: sources.value[1].sizes,
-            srcset: sources.value[1].srcset
-          })]
-        : []),
-      h('img', {
-        ref: imgEl,
-        ..._base.attrs.value,
-        ...process.server ? { onerror: 'this.setAttribute(\'data-error\', 1)' } : {},
-        ...imgAttrs,
-        src: sources.value[0].src,
-        sizes: sources.value[0].sizes,
-        srcset: sources.value[0].srcset
-      })
-    ])
+    return () =>
+      h('picture', { key: sources.value[0].src }, [
+        ...sources.value.slice(0, -1).map((source) => {
+          return h('source', {
+            type: source.type,
+            sizes: source.sizes,
+            srcset: source.srcset,
+            src: source.src
+          })
+        }),
+        h('img', {
+          ref: imgEl,
+          ..._base.attrs.value,
+          ...(process.server ? { onerror: "this.setAttribute('data-error', 1)" } : {}),
+          ...imgAttrs,
+          src: sources.value[lastSourceIndex.value].src,
+          sizes: sources.value[lastSourceIndex.value].sizes,
+          srcset: sources.value[lastSourceIndex.value].srcset
```

Comment:
```suggestion
          src: sources.value[0].src,
          sizes: sources.value[0].sizes,
          srcset: sources.value[0].srcset
```

Reply:
This would flip it the wrong way. The array is order [...format, legacyFormat]. So avif,webp would return [avif,webp,png]. Right?

Reply:
I think we want the `<img>` rendered to be the non-legacy version. So `[0]` is `avif` in your example. The rest (`webp`, `png`) are the next two, which are rendered above this.

Reply:
But the `<img>` is served as the fallback. So it should be the legacyFormat?

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture#the_type_attribute

Reply:
but then it seems we were previously (wrongly?) preloading the legacy format image

Reply:
I think yes